File: banco_problemas_aplicados/nivel_basico/soluciones/java/Problema01.java
/*
 * Problema 01 - Suma de Muestras de Corriente ADC
 * double sumaCorrientes(double[] arr, int tam)  -> sumatoria total (amperios)
 */
import java.util.Scanner;
import java.util.Locale;

public class Problema01 {

    static double sumaCorrientes(double[] arr, int tam) {
        double suma = 0.0;
        for (int i = 0; i < tam; i++) {
            suma = suma + arr[i];
        }
        return suma;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Ingrese el numero de muestras: ");
        int tam = sc.nextInt();

        double[] arr = new double[tam];
        for (int i = 0; i < tam; i++) {
            System.out.print("Muestra " + i + " (A): ");
            arr[i] = sc.nextDouble();
        }

        System.out.printf(Locale.US, "Suma total de corrientes: %.2f A%n", sumaCorrientes(arr, tam));
        sc.close();
    }
}

File: banco_problemas_aplicados/nivel_basico/soluciones/cpp/problema01.cpp
/*
 * Problema 01 - Suma de Muestras de Corriente ADC
 * double sumaCorrientes(double Arr[], int tam)  -> sumatoria total (amperios)
 */
#include <iostream>
#include <iomanip>
using namespace std;

double sumaCorrientes(double arr[], int tam) {
    double suma = 0.0;
    for (int i = 0; i < tam; i++) {
        suma = suma + arr[i];
    }
    return suma;
}

int main() {
    int tam;
    cout << "Ingrese el numero de muestras: ";
    cin >> tam;

    double* arr = new double[tam];
    for (int i = 0; i < tam; i++) {
        cout << "Muestra " << i << " (A): ";
        cin >> arr[i];
    }

    cout << fixed << setprecision(2);
    cout << "Suma total de corrientes: " << sumaCorrientes(arr, tam) << " A" << endl;

    delete[] arr;
    return 0;
}
